### PR TITLE
update diligent-core in order to make diligent-tools recipe

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **diligent-core/2.51**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
@uilianries @SSE4 @SpaceIm 
please review. I want to post diligent-tools recipe

File: recipes/diligent-core/all/conanfile.py [hunk 1]
```diff
@@ -154,8 +154,17 @@ def package_info(self):
 
         self.cpp_info.libs = tools.collect_libs(self)
         self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore"))
-        # fake target. Needed for DiligentFx to handle paths like ../../../DiligentCore
         self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Common", "interface"))
+        self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Primitives", "interface"))
```
Comment (SSE4) [inline recipes/diligent-core/all/conanfile.py:158]: wow, that's a lot new includes

**Review by SSE4 — APPROVED** [review]

File: recipes/diligent-core/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/diligent-core/all/conanfile.py:158]: It's weird. It brings many generic headers names without namespace folder.

**Review by SpaceIm — APPROVED** [review]

File: recipes/diligent-core/all/conanfile.py (hunk 1 above)
Comment (AndreyMlashkin) [reply recipes/diligent-core/all/conanfile.py:158]: Yes, it's really confusing...
But look for example on https://github.com/DiligentGraphics/DiligentTools/blob/master/TextureLoader/include/pch.h

These files are referenced without full path...

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**